presto-hive/src/main/java/com/facebook/presto/hive/HiveMaterializedViewUtils.java

```diff
@@ -95,17 +96,24 @@ public static void validateMaterializedViewPartitionColumns(
                         .orElseThrow(() -> new TableNotFoundException(baseTableName)))
                 .collect(toImmutableList());
 
-        Map<SchemaTableName, List<Column>> baseTablePartitions = baseTables.stream()
+        Map<Table, List<Column>> baseTablePartitions = baseTables.stream()
                 .collect(toImmutableMap(
-                        table -> new SchemaTableName(table.getDatabaseName(), table.getTableName()),
+                        table -> table,
                         Table::getPartitionColumns));
 
-        for (SchemaTableName baseTable : baseTablePartitions.keySet()) {
-            if (!isCommonPartitionFound(baseTable, baseTablePartitions.get(baseTable), viewPartitions, viewToBaseColumnMap)) {
+        for (Table baseTable : baseTablePartitions.keySet()) {
+            SchemaTableName schemaBaseTable = new SchemaTableName(baseTable.getDatabaseName(), baseTable.getTableName());
+            if (!isCommonPartitionFound(schemaBaseTable, baseTablePartitions.get(baseTable), viewPartitions, viewToBaseColumnMap)) {
```

Comment:
Maybe I missed something. 
This viewToBaseColumnMap doesn't include the nullable columns but baseTable could be a table on the outer join side. Then will isCommonPartitionFound will return false for this case?

Reply:
We expect both t1.ds and t2.ds to be included in materialized view in this case. In future, we can remove this restriction, but needs more cases to handle. I wrote this in the PR summary